tests/unit/nupic/data/functionsource_test.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2013, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""
+Unit tests for functionsource.
+"""
+
+import unittest2 as unittest
+import pickle
+
+from nupic.data import FunctionSource
+
+def dataFunction(stat):
+  ret = {"reset": 0, "sequence": 0, "data": 0}
+  if stat is not None:
+    val = stat.get('val', 0) + 1
+    ret["val"] = stat["val"] = val
+  return ret
+
+class FunctionSourceTest(unittest.TestCase):
+  def testFunctionSource(self):
+    fs = FunctionSource(dataFunction, state=None, resetFieldName=None, sequenceIdFieldName=None)
+    self.assertIsNotNone(fs)
+    r = fs.getNextRecordDict()
+    self.assertIsNotNone(r)
+
+  def testFunctionSourceReset(self):
```

Comment:
you can remove "FunctionSource" from each test method name since they are all inside the class `FunctionSourceTest`.